telemetrylib/Telemetry.cpp

```diff
@@ -0,0 +1,75 @@
+//
+// Created by Mingcan Li on 2/3/23.
+// Commented by ChatGPT
+//
+#include "Telemetry.h"
+
+// Default constructor
+Telemetry::Telemetry() {}
+
+// Constructor with commChannels initialization
+Telemetry::Telemetry(std::vector<DTI *> commChannels) {
+    // Loop through all communication channels and add them to the comm vector
+    for (int i = 0; i < commChannels.size(); i++) {
+        this->comm.push_back(commChannels[i]);
+        // Connect the signal for connection status changes to the comChannelChanged() slot
+        connect(comm[i], SIGNAL(connectionStatusChanged()), this, SLOT(comChannelChanged()));
+    }
+    // Output the number of initialized communication channels to console
+    qDebug() << "comm channels initialized: " << comm.size();
+}
+
+// Send data through the current communication channel
+void Telemetry::sendData(QByteArray bytes) {
+    qDebug() << "send data current comm channel: " << commChannel;
+    // Check if there is a current communication channel and if the resend queue is not busy
+    if (commChannel != -1 && !resendQueue.isBusy()) {
+        // Send data through the current communication channel
+        comm[commChannel]->sendData(bytes);
+    } else {
+        qDebug()<<"adding to queue from telemetry";
+        // Add data to the resend queue if there is no current communication channel or the resend queue is busy
+        resendQueue.addToQueue(bytes);
+    }
+}
```

Comment:
Let's have this take in a time parameter since we're getting the current time in backendprocesses for setting the `tstamp_xx` fields in the byte array. Then, that time should be passed as an argument to `sendData()` and `addToQueue()` so that we don't have to have special checks in those functions for whether or not a timestamp was passed.